vhost/src/vhost_user/message.rs

```diff
@@ -907,6 +913,56 @@ impl VhostUserMsgValidator for VhostUserLog {
     }
 }
 
+enum_value! {
+    /// Direction of state transfer for migration
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+    pub enum VhostTransferStateDirection: u32 {
+        /// Outgoing migration: Transfer state from back-end to front-end
+        SAVE = 0,
+        /// Incoming migration: Transfer state from front-end to back-end
+        LOAD = 1,
+    }
+}
+
+enum_value! {
+    /// Migration phases during which state transfer can occur
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+    pub enum VhostTransferStatePhase: u32 {
+        /// The device (and all its vrings) are stopped
+        STOPPED = 0,
+    }
+}
+
+/// Query/send virtio-fs migration state
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct VhostUserTransferDeviceState {
+    /// Direction of state transfer (save/load)
+    pub direction: u32,
+    /// Migration phase during which the transfer takes place
+    pub phase: u32,
+}
+
+// SAFETY: Safe because all fields of VhostUserTransferDeviceState are POD.
```

Comment:
POD is only one requirement. Since we seem to got this wrong in a few places we should also mention the packed requirement here IMHO. What do you think?

Reply:
Done